korge/src/commonMain/kotlin/com/soywiz/korge/view/RepeatedImageView.kt

```diff
@@ -0,0 +1,206 @@
+package com.soywiz.korge.view
+
+import com.soywiz.kds.*
+import com.soywiz.kds.iterators.fastForEach
+import com.soywiz.kmem.*
+import com.soywiz.korge.internal.KorgeInternal
+import com.soywiz.korge.render.RenderContext
+import com.soywiz.korge.render.TexturedVertexArray
+import com.soywiz.korim.bitmap.Bitmap
+import com.soywiz.korim.bitmap.Bitmaps
+import com.soywiz.korim.bitmap.BmpSlice
+import com.soywiz.korma.geom.Point
+import com.soywiz.korma.geom.Rectangle
+import kotlin.math.max
+import kotlin.math.min
+
+inline fun Container.repeatedImageView(
+    bitmap: BmpSlice,
+    repeatX: Boolean = false,
+    repeatY: Boolean = false,
+    smoothing: Boolean = true,
+    block: @ViewDslMarker RepeatedImageView.() -> Unit = {}
+) = RepeatedImageView(bitmap, smoothing, repeatX, repeatY).addTo(this, block)
+
+open class RepeatedImageView(
```

Comment:
Do you think RepeatedImageView could be replaced by a TileMap with a single cell + a single image TileSet like in this sample?

https://github.com/korlibs/korge-samples/blob/231fa10e28e44a0e1415aaa1d031c11f382a55bd/samples/tiled-background/src/commonMain/kotlin/main.kt#L11-L12

Reply:
Yes, I guess this would be possible. In fact I created RepeatedImageView.kt by making a copy of TileMap.kt and throwing out everything what I did not need.

Reply:
I just tried to use a TileMap and now I remember why I did the RepeatedImageView as a "strip down" of a TileMap. (That was quite some time ago...) It is impossible to change the image inside the TileSet without recreating the whole object. But that is needed in setFrame function. So it would be not very efficient to have animated images for the layers of the parallax background (which I do not yet have but which might be an option for animated backgrounds :)

Reply:
I see. Maybe we can create an abstract class or extract the functionality to some internal methods so the code is not duplicated.
Not high priority, and we can skip this for now for this PR, but would be nice to have at some point to avoid those two implementations evolving differently or not improving both of them when updating one.
What do you think?

Reply:
I fully agree. I can check how to "merge" RepeatedImageView with TileMap class for that.

Another question: I looked into TileMap for "repeating" the Bitmap because that was already implemented there (for the tiles). I did not succeed to add "repeating" of the Bitmap in e.g. BaseImage when I first looked into it. Which is quite some while ago. Now I have a bit better understanding of the Korge sources on that topic.
What do you think. Should we try to add "repeatX" and "repeatY" to BaseImage? Or should the "repeat" feature be kept separate in the TileMap class?

Reply:
I just looked into the code again and have an idea how I can merge RepeatedImageView back to TileMap. I have the idea to create an abstract base class "BaseTileMap" and derive from that TileMap and something like "SingleTile" (as replacement for RepeatedImageView).

While looking into that I realized that RepeatedImageAnimationView can also be "merged" back to ImageAnimationView class. There I would also save a lot duplicated code.

I have time to look into it this evening...

Reply:
Cool. That's a nice idea!